src/vivarium_ciff_sam/components/observers.py

```diff
@@ -84,7 +93,15 @@ def get_state_function(state: Union[str, bool, List]) -> Callable:
                                  {'covered': True, 'uncovered': False})
 
         if self.by_x_factor:
-            setup_stratification('x_factor.exposure', True, 'x_factor', ('cat2', 'cat1'))
+            setup_stratification('x_factor.exposure', True, 'x_factor', results.DICHOTOMOUS_RISK_STATES)
+
+        if self.by_maternal_malnutrition:
+            setup_stratification(
+                f'{data_keys.MATERNAL_MALNUTRITION.name}.exposure',
+                True,
```

Comment:
In the class constructor, self.by_maternal_malnutrition is initialized by setting it != "False" (str) instead of a bool where the in the setup_stratification method, it uses a bool.  Is there a functionality this provides or is it easier to define this way?

Reply:
The reason we do it this way is that these components are defined by parsing the config file which takes the input arguments as strings and can't handle bools. I could either convert them to bools in the individual observer `__init__` calls or centrally here in the `ResultsStratifier`.